### DISCUSSION TITLE
java.lang.RuntimeException: Unable to open default system browser

### DISCUSSION THREAD

**Discussion by lightningdave** [dbeaver/dbeaver#18191] [Q&A] [answered]
Hi, 

I have dbeaver installed on pop_os.
Now I would like to connect to an azure resources using aad/mfa.

I get the exception: 

    Failed to authenticate the user  in Active Directory (Authentication=ActiveDirectoryInteractive).
      java.lang.RuntimeException: Unable to open default system browser
      java.lang.RuntimeException: Unable to open default system browser
        Unable to open default system browser
        Unable to open default system browser

I cant select a internal browser (is empty).
I don' t know how to point to an external browser (I have firefox installed as system default).
Tried to point to Firefox, but cant seem to get that seemingly simple setting to work...

![image](https://user-images.githubusercontent.com/92384389/201295790-399cbe5b-200e-43a5-99fe-e86bea696738.png)

Thanks

**Comment by E1izabeth** [comment, marked as answer]
Have you tried to press `New..` button and specify the path to your browser?
Could you try again please and send the log files after that? You can find instructions here https://github.com/dbeaver/dbeaver/wiki/Log-files

**Reply by lightningdave** [reply]
Hi thanks for responding,

'which firefox' results in this path: /snap/bin/firefox

so I have put this in as external browser:
![image](https://user-images.githubusercontent.com/92384389/201350725-db38afd3-53c7-4c77-89ac-a4b7691c0546.png)

Still get the exception 'Unable to open default system browser'.

Please find the logs attached.
[dbeaver-logs.txt](https://github.com/dbeaver/dbeaver/files/9990585/dbeaver-logs.txt)

**Reply by Destrolaric** [reply]
Can you try to install firefox via dpkg, I think it may be related to snap sandboxing.

**Reply by lightningdave** [reply]
I have removed the snap version and installed a .deb version (followed this [link](https://www.omgubuntu.co.uk/2022/04/how-to-install-firefox-deb-apt-ubuntu-22-04))

Now I have this. 'Location not a valid path'.

which firefox gives: /usr/bin/firefox

Seems to be a link file.
I can look inside the link and follow the path to elsewhere (/usr/lib/firefox/firefox), but the still not loading browser to do the MFA stuff.

![image](https://user-images.githubusercontent.com/92384389/201367911-8d4bb624-e2a4-43a7-b626-ce1736d179ee.png)

**Reply by lightningdave** [reply]
Would anyone be so kind to post screenshot of what a working configuration looks like?
Connecting to an Azure SQL DB with MFA, and that there browser starts?

Just curious if it is me, or the whole thing just doesnt work.

I have tried on my other linux system as well, tried installing firefox and chrome via .deb.
No succes,... so perhaps its something else?

com.microsoft.aad.msal4j.MsalClientException: Unable to open default system browser

Thanks

**Reply by underq** [reply]
Hi @lightningdave, I've the same error, you are not alone :hugs:

**Reply by lightningdave** [reply]
Thanks for the confirmation @underq !
I was starting to doubt if I was asking something dumb.
I have done all the steps proposed in this thread, but still dbeaver doesn't launch on both my pop_os and Ubuntu installs.
Tried with chrome and firefox, installed through both flatpak, apt and .deb
Nothing seems to work ...

**Reply by Destrolaric** [reply]
@lightningdave Sorry for waiting, I think it is not related to browser sandboxing, it's possible that it was caused by the snap of DBeaver itself. You can try to install DBeaver via dpkg.

**Reply by lightningdave** [reply]
Yesss!! That did the trick! :smiley: 
Finally working!

Thanks

**Comment by lightningdave** [comment]
The awnser was to install DBeaver though dpkg .deb
And not through flatpak.
After reinstalling it through dpkg, the system browser was directly loading the first time i called for MFA.
Thanks @Destrolaric  for the help!
